Core/GDCore/Extensions/Metadata/DependencyMetadata.h

```diff
@@ -132,6 +133,18 @@ class GD_CORE_API DependencyMetadata {
     return extraData;
   }
 
+  void CopyFrom(const DependencyMetadata& dependencyMetadata) {
+    name = dependencyMetadata.name;
+    exportName = dependencyMetadata.exportName;
+    version = dependencyMetadata.version;
+    dependencyType = dependencyMetadata.dependencyType;
+    extraData = dependencyMetadata.extraData;
+    onlyIfSomeExtraSettingsNonEmpty =
+        dependencyMetadata.onlyIfSomeExtraSettingsNonEmpty;
+    onlyIfOtherDependencyIsExported =
+        dependencyMetadata.onlyIfOtherDependencyIsExported;
```

Comment:
Give a try at just `*this = dependencyMetadata;` or `operator=(dependencyMetadata)`.
We should not need to redefine what a copy is, as the compiler is generating this code for us in the form of the assignment operator (and the copy constructor).